tests/select_for_update/tests.py

```diff
@@ -61,6 +61,24 @@ def end_blocking_transaction(self):
         self.new_connection.rollback()
         self.new_connection.set_autocommit(True)
 
+    def start_no_key_blocking_transaction(self):
+        self.new_connection.set_autocommit(False)
+        # Start a blocking transaction. At some point,
+        # end_blocking_transaction() should be called.
+        self.cursor = self.new_connection.cursor()
+        sql = 'SELECT * FROM %(db_table)s %(for_update)s;' % {
+            'db_table': City._meta.db_table,
+            'for_update': self.new_connection.ops.for_update_sql(no_key=True),
+        }
```

Comment:
Can we test this feature without raw SQLs?

Reply:
Yes, in fact, there is a "better" way of testing this kind of lock:
If I lock a `City` with `select_for_update` (without `no_key` option) and then I try to `.save` a `Person` referencing that city the test would hang forever.
The approach would be "if it does not hang, it's OK", but if it hangs... If it hangs then we can't raise anything nor continue executing the rest of the test suite. (Note that I cannot do something like `nowait=True` to avoid deadlocks since `INSERT` hasn't got that option)

What this test ensures is that `SELECT FOR NO KEY UPDATE` works at least as `SELECT FOR UPDATE` does, for which I created this method that basically does the same as `start_blocking_transaction`. What if I merged these functions into the old one? Would that be OK?